strider/fetcher.py

```diff
@@ -152,42 +143,81 @@ async def generate_plan(self):
 
         self.logger.debug("Generating plan")
         # Generate traversal plan
-        plans = await generate_plans(
+        plans, kps = await generate_plans(
             self.qgraph,
             kp_registry=registry,
             logger=self.logger)
+        self.kps = kps
 
         if len(plans) == 0:
             self.logger.error("Could not find a plan to traverse query graph")
             raise NoAnswersError()
 
         self.plan = plans[0]
 
+        # extract KP preferred prefixes from plan
+        self.kp_preferred_prefixes = dict()
+        for _kps in kps.values():
+            for kp in _kps:
+                url = kp["url"][:-5] + "meta_knowledge_graph"
+                # url = kp["url"][:-5] + "metadata"
+                try:
+                    async with httpx.AsyncClient() as client:
+                        response = await client.get(url)
+                except httpx.ConnectError as err:
+                    self.logger.warning(
+                        "Unable to get meta knowledge graph for KP %s: %s",
+                        kp["url"],
+                        str(err),
+                    )
+                    self.kp_preferred_prefixes[kp["url"]] = dict()
+                    continue
+                try:
+                    meta_kg = response.json()
+                except JSONDecodeError as err:
+                    self.logger.warning(
+                        "Unable to get meta knowledge graph for KP %s: %s",
```

Comment:
I would maybe make this a more descriptive error message. We were able to get the meta knowledge graph, but it came back malformed.

Reply:
Modified in 62ea25d